.github/workflows/quarkus-nightly.yml

```diff
@@ -0,0 +1,78 @@
+name: Quarkus Nightly
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - quarkus-nightly
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: quarkus-nightly-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  rebase-quarkus-nightly-branch:
+    name: Rebase quarkus-nightly branch
+    if: github.event_name != 'schedule' || github.repository == 'keycloak/keycloak'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: quarkus-nightly
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@github.com"
+
+      - name: Fetch and rebase
+        run: |
+          git fetch origin quarkus-nightly
+          git rebase origin/main
```

Comment:
This would break I reckon if there are any changes to `pom.xml`. Might be better to do something like:

```
git checkout main
git checkout -b new-quarkus-nightly
mvn versions:update-property -Dproperty=quarkus.version -DnewVersion=999-SNAPSHOT
git commit -m "Set quarkus version"
git cherry-pick <cherry-pick any additional commits in quarkus-nightly, except the updated to quarkus.version above>
```

Reply:
This turned out to be a little bit more complex, please take a look. I can move it into a separate script if necessary.

Thanks.